frontend/public/reducers/k8s.ts

```diff
@@ -1,89 +1,17 @@
 
 import { ActionType, K8sAction } from '../actions/k8s';
-import { getQN, referenceForModel } from '../module/k8s/k8s';
-import { K8sResourceKind, K8sKind } from '../module/k8s/types';
+import { referenceForModel } from '../module/k8s/k8s';
+import { K8sKind } from '../module/k8s/types';
 import { allModels } from '../module/k8s/k8s-models';
 import { getNamespacedResources } from '../actions/ui';
 
-const moreRecent = (a, b) => {
-  const metaA = a.get('metadata').toJSON();
-  const metaB = b.get('metadata').toJSON();
-  if (metaA.uid !== metaB.uid) {
-    return new Date(metaA.creationTimestamp) > new Date(metaB.creationTimestamp);
-  }
-  return parseInt(metaA.resourceVersion, 10) > parseInt(metaB.resourceVersion, 10);
-};
-
-const removeFromList = (list, resource) => {
-  const qualifiedName = getQN(resource);
-  // eslint-disable-next-line no-console
-  console.log(`deleting ${qualifiedName}`);
-  return list.delete(qualifiedName);
-};
-
-const updateList = (list: ImmutableMap<string, any>, nextJS: K8sResourceKind) => {
-  const qualifiedName = getQN(nextJS);
-  const current = list.get(qualifiedName);
-  const next = fromJS(nextJS);
-
-  if (!current) {
-    return list.set(qualifiedName, next);
-  }
-
-  if (!moreRecent(next, current)) {
-    return list;
-  }
-
-  // TODO: (kans) only store the data for things we display ...
-  //  and then only do this comparison for the same stuff!
-  if (
-    current
-      .deleteIn(['metadata', 'resourceVersion'])
-      .equals(next.deleteIn(['metadata', 'resourceVersion']))
-  ) {
-    // If the only thing that differs is resource version, don't fire an update.
-    return list;
-  }
-
-  return list.set(qualifiedName, next);
-};
-
-const loadList = (oldList, resources) => {
-  const existingKeys = new Set(oldList.keys());
-  return oldList.withMutations((list) => {
-    (resources || []).forEach((r) => {
-      const qualifiedName = getQN(r);
-      existingKeys.delete(qualifiedName);
-      const next = fromJS(r);
-      const current = list.get(qualifiedName);
-      if (!current || moreRecent(next, current)) {
-        list.set(qualifiedName, next);
-      }
-    });
-    existingKeys.forEach((k) => {
-      const r = list.get(k);
-      const metadata = r.get('metadata').toJSON();
-      if (!metadata.deletionTimestamp) {
-        // eslint-disable-next-line no-console
-        console.warn(`${metadata.namespace}-${metadata.name} is gone with no deletion timestamp!`);
-      }
-      list.delete(k);
-    });
-  });
-};
-
 export type K8sState = ImmutableMap<string, any>;
 
 export default (state: K8sState, action: K8sAction): K8sState => {
   if (!state) {
     return fromJS({ RESOURCES: { inFlight: false, models: ImmutableMap<string, K8sKind>() } });
   }
-  // const {k8sObjects, id} = action;
-  // const list: ImmutableMap<string, any> = state.getIn([id, 'data']);
-
-  let newList;
 
   switch (action.type) {
     case ActionType.GetResourcesInFlight:
```

Comment:
What was the reasoning again for not moving all of the k8s reducer cases over?

Reply:
@andrewballantyne the reducers left here were not explicitly required for the utilities we moved, and have the non-pure Redux anti-pattern we discussed here: https://coreos.slack.com/archives/C02EQERSV3N/p1637681777395300?thread_ts=1637680014.387100&cid=C02EQERSV3N

I think it's going to be much more of a challenge to migrate these. I think that should be another PR.